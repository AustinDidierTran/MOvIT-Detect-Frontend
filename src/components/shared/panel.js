/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import '../../styles/panel.css';
import '../../styles/overwrite.css';

import React, { Component } from 'react';

import { Panel } from 'primereact/components/panel/Panel';
import PropTypes from 'prop-types';

export default class CustomPanel extends Component {
  static propTypes = {
    element: PropTypes.element.isRequired,
    title: PropTypes.string.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      panelCollapsed: true,
    };
  }

  render() {
    const header = (
      <span className="header">
        {this.props.title}
      </span>
    );
    return (
      <div className="container">
        <div className="card">
          <Panel
            header={header}
            toggleable
            collapsed={this.state.panelCollapsed}
            onCollapse={() => this.setState({ panelCollapsed: !this.state.panelCollapsed })
            }
          >
            {this.props.element}
          </Panel>
        </div>
      </div>
    );
  }
}
