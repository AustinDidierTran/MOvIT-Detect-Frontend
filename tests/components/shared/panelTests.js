/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import { Panel } from 'primereact/components/panel/Panel';
import PropTypes from 'prop-types';
import React from 'react';
import toJson from 'enzyme-to-json';
import CustomPanel from '../../../src/components/shared/panel';

Enzyme.configure({ adapter: new Adapter() });

describe('CustomPanel Tests', () => {
  let wrapper;
  const props = {
    element: <div>Test</div>,
    title: 'This is a test',
  };

  beforeEach(() => {
    wrapper = shallow(<CustomPanel {...props} />);
  });

  it('should have proptypes', () => {
    const actualValue = CustomPanel.propTypes;

    const expectedValue = {
      element: PropTypes.element.isRequired,
      title: PropTypes.string.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should expand the panel', () => {
    wrapper.setState({ panelCollapsed: true });

    wrapper.find(Panel).simulate('collapse');

    expect(wrapper.state('panelCollapsed')).toEqual(false);
  });

  it('should collapse the panel', () => {
    wrapper.setState({ panelCollapsed: false });

    wrapper.find(Panel).simulate('collapse');

    expect(wrapper.state('panelCollapsed')).toEqual(true);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
