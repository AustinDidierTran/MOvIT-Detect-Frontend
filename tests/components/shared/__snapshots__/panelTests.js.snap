// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPanel Tests should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="card"
  >
    <Panel
      className={null}
      collapsed={true}
      header={
        <span
          className="header"
        >
          This is a test
        </span>
      }
      id={null}
      onCollapse={[Function]}
      onExpand={null}
      style={null}
      toggleable={true}
    >
      <div>
        Test
      </div>
    </Panel>
  </div>
</div>
`;
